### PR TITLE
Remove Dependabot and Reconfigure Renovate bot

### PR DESCRIPTION
**Pull request by yashvardhannanavati**
Refers to CLOUDDST-19743

Custom config for IIB. Reference: https://docs.renovatebot.com/configuration-options/

### CODE REVIEW

**Review by xDaile — DISMISSED** [review]

**Review by chandwanitulsi — APPROVED** [review]

**Review by zanssa — APPROVED** [review]

File: .github/renovate.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "enabled": true,
+  "enabledManagers": ["docker-compose", "dockerfile", "github-actions", "pip_requirements", "pyenv"],
```
Comment (lipoja) [inline .github/renovate.json:3]: Are we sure that default `dockerfile` manager is matching our dockerfiles? 
https://docs.renovatebot.com/modules/manager/dockerfile/#default-config

From that regexp I think it is not. Maybe we should add `fileMatch` for `dockerfile` as well?

**Review by lipoja — CHANGES_REQUESTED** [review]

File: .github/renovate.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+  "branchConcurrentLimit": 5,
+  "docker-compose": {
+    "fileMatch": ["podman-compose.yml$"]
```
[the remark this answers is not in the document]
Comment (yashvardhannanavati) [reply .github/renovate.json:6]: So, `fileMatch` appends to the existing patterns they have by default. If you check the comment here: https://github.com/release-engineering/iib/pull/535, everything we want to track is being tracked by default except for `podman-compose.yml`. Hence, I only added the missing file here.

File: .github/renovate.json (hunk 1 above)
Comment (yashvardhannanavati) [reply .github/renovate.json:3]: Yes, it is matching the dockerfiles : https://github.com/release-engineering/iib/pull/535

**Review by zanssa — CHANGES_REQUESTED** [review]
Inspired by a discussion about having specific commits as it helps when looking at the commit history in the future, shouldn't this change be splitted into multiple commits as I don't see it reflecting the removal of the old method to handle the dependencies update?

**Comment by yashvardhannanavati** [comment]
> Inspired by a discussion about having specific commits as it helps when looking at the commit history in the future, shouldn't this change be splitted into multiple commits as I don't see it reflecting the removal of the old method to handle the dependencies update?

Yes, you're absolutely right. I split it into two commits.

**Comment by zanssa** [comment]
Giving this more thought, I am not sure how pyenv manager would work. Maybe you can help me here @yashvardhannanavati 

Looking at this MR #355, all the work needed in CLOUDDST-10861, and the tasks that were triggered by that change, do we need to keep pyenv enabled in the configuration? 

Maybe we should enable this in stages and based on that enable it for others managers. thoughts?

**Comment by yashvardhannanavati** [comment]
> Giving this more thought, I am not sure how pyenv manager would work. Maybe you can help me here @yashvardhannanavati 
> 
> 
> 
> Looking at this MR #355, all the work needed in CLOUDDST-10861, and the tasks that were triggered by that change, do we need to keep pyenv enabled in the configuration? 
> 

> 
> 
> Maybe we should enable this in stages and based on that enable it for others managers. thoughts? 

So as of today, all the managers are enabled since we have the default config. The reason I didn't disable it was thinking that we can always close the PR if it ever creates one since automerge is not enabled. But I do agree that we updated the pyenv file manually when we dropped support for 3.6 and upgraded to 3.8. So I am open to dropping pyenv from that list and that will disable it. Thoughts?

**Review by xDaile — APPROVED** [review]

**Review by lipoja — APPROVED** [review]

**Comment by yashvardhannanavati** [comment]
@zanssa We discussed this in the guild meeting and thought it might be better to keep the `pyenv` manager along with the others. Could you please share if you have any concerns? :)

**Review by chandwanitulsi — APPROVED** [review]

**Pull request merged**